lib/SILOptimizer/Mandatory/OSLogOptimization.cpp

```diff
@@ -0,0 +1,631 @@
+          numberDecl == astCtx.getUInt64Decl() ||
+          numberDecl == astCtx.getBoolDecl());
+}
+
+/// Return true if and only if the given SIL type represents a String or
+/// a Stdlib or builtin integer type.
+static bool isIntegerOrStringType(SILType silType, ASTContext &astContext,
+                                  bool excludeSILAddresses = false) {
+  if (excludeSILAddresses && silType.isAddress()) {
+    return false;
+  }
+
+  if (silType.is<BuiltinIntegerType>()) {
+    return true;
+  }
+
+  auto *nominalDecl = silType.getNominalOrBoundGenericNominal();
+  if (!nominalDecl) {
+    return false;
+  }
+
+  return (nominalDecl == astContext.getStringDecl()) ||
+         isStdlibIntegerOrBoolDecl(nominalDecl, astContext);
+}
+
+/// If the given instruction is a call to the compiler-intrinsic initializer
+/// of String that accepts string literals, return the called function.
+/// Otherwise, return nullptr.
+static SILFunction *getStringMakeUTF8Init(SILInstruction *inst) {
+  auto *apply = dyn_cast<ApplyInst>(inst);
+  if (!apply)
+    return nullptr;
+
+  auto *callee = apply->getCalleeFunction();
+  if (!callee || !callee->hasSemanticsAttr("string.makeUTF8"))
+    return nullptr;
+  return callee;
+}
+
+/// Decide if the given instruction (which could possibly be a call) should
+/// be constant evaluated.
+///
+/// \returns true iff the given instruction is not a call or if it is, it calls
+/// a known stdlib function that the interpreter supports or calls a os log
+/// overlay function annotated with a semantics attribute.
+static bool shouldAttemptEvaluation(SILInstruction *inst,
+                                    ConstExprStepEvaluator &stepEval) {
+  auto *apply = dyn_cast<ApplyInst>(inst);
+  if (!apply)
+    return true;
+
+  auto calleeFun = apply->getCalleeFunction();
+  if (!calleeFun)
+    return false;
+
+  return stepEval.isKnownFunction(calleeFun) ||
+         calleeFun->hasSemanticsAttrThatStartsWith("oslog");
+}
+
+/// Skip or evaluate the given instruction based on the evaluation policy and
+/// handle errors. The policy is to evaluate all non-apply instructions as well
+/// as apply instructions that either invoke a known stdlib function or an os
+/// log specific function that constructs compile-time constants
+/// (like format string). Every other function call is skipped.
+/// This includes calls that manipulate runtime values such as the arguments
+/// (i.e, interpolated expressions) or the raw byte buffer.
+static std::pair<Optional<SILBasicBlock::iterator>, Optional<SymbolicValue>>
+evaluateOrSkip(ConstExprStepEvaluator &stepEval,
+               SILBasicBlock::iterator instI) {
+  auto *inst = &(*instI);
+
+  // Note that skipping a call conservatively approximates its effects on the
+  // interpreter state.
+  if (shouldAttemptEvaluation(inst, stepEval)) {
+    return stepEval.tryEvaluateOrElseMakeEffectsNonConstant(instI);
+  }
+  return stepEval.skipByMakingEffectsNonConstant(instI);
```

Comment:
I assume this will mark the alloc_stack of the string as "unknown" if this is an unknown function which (potentially) modifies the string in the alloc_stack

Reply:
Yes. It will mark the content of the address returned by alloc_stack as unknown. The address itself is a valid symbolic value, but it contents will be unknown.